flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetTableSource.java

```diff
@@ -447,8 +453,16 @@ private String getColumnName(BinaryComparison comp) {
 
 	@Nullable
 	private Tuple2<Column, Comparable> extractColumnAndLiteral(BinaryComparison comp) {
-		TypeInformation<?> typeInfo = getLiteralType(comp);
 		String columnName = getColumnName(comp);
+		TypeInformation<?> typeInfo = null;
+		try {
+			String[] columnPath = columnName.split("\\.");
+			Type type = parquetSchema.getType(columnPath);
+			typeInfo = ParquetSchemaConverter.convertParquetTypeToTypeInfo(type);
+		} catch (InvalidRecordException e) {
```

Comment:
Why catch this exception? This should never happen?
If it will happen, use `parquetSchema.containsPath` first?

Reply:
It is not needed.